tests/components/device_tracker/test_init.py

```diff
@@ -234,6 +234,8 @@ async def test_discover_platform(
 ) -> None:
     """Test discovery of device_tracker demo platform."""
     await async_setup_component(hass, "homeassistant", {})
+    await async_setup_component(hass, device_tracker.DOMAIN, {})
+    await hass.async_block_till_done()
```

Comment:
Needing this here is bug. But the regression wasn't introduced in this PR, its only revealing the race.

I fixed it in https://github.com/home-assistant/core/pull/114101